### PR TITLE
#450 added seal wrap parameter to hashivault secret engine

### PR DESCRIPTION
**Pull request by emoryotott19**

### CODE REVIEW

**Comment by TerryHowe** [comment]
```
ansible/modules/hashivault/hashivault_secret_engine.py:228:5: E303 too many blank lines (2)
ansible/modules/hashivault/hashivault_secret_engine.py:234:121: E501 line too long (139 > 120 characters)
ansible/modules/hashivault/hashivault_secret_engine.py:244:121: E501 line too long (13[9](https://github.com/TerryHowe/ansible-modules-hashivault/actions/runs/6433936602/job/17473108807?pr=452#step:5:10) > 120 characters)
```

**Comment by TerryHowe** [comment]
Closes: #450

**Comment by TerryHowe** [comment]
```
 An exception occurred during task execution. To see the full traceback, use -vvv. The error was: TypeError: configure() got an unexpected keyword argument 'seal_wrap'
  fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"/home/runner/.ansible/tmp/ansible-tmp-1696617947.625752-9286-215266148506769/AnsiballZ_hashivault_secret_engine.py\", line 107, in <module>\n    _ansiballz_main()\n  File \"/home/runner/.ansible/tmp/ansible-tmp-1696617947.625752-9286-215266148506769/AnsiballZ_hashivault_secret_engine.py\", line 99, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/runner/.ansible/tmp/ansible-tmp-1696617947.625752-9286-215266148506769/AnsiballZ_hashivault_secret_engine.py\", line 47, in invoke_module\n    runpy.run_module(mod_name='ansible.modules.hashivault.hashivault_secret_engine', init_globals=dict(_module_fqn='ansible.modules.hashivault.hashivault_secret_engine', _modlib_path=modlib_path),\n  File \"/opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/runpy.py\", line 225, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/runpy.py\", line 97, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/runpy.py\", line 87, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_hashivault_secret_engine_payload_n8x91xqz/ansible_hashivault_secret_engine_payload.zip/ansible/modules/hashivault/hashivault_secret_engine.py\", line 295, in <module>\n  File \"/tmp/ansible_hashivault_secret_engine_payload_n8x91xqz/ansible_hashivault_secret_engine_payload.zip/ansible/modules/hashivault/hashivault_secret_engine.py\", line 120, in main\n  File \"/tmp/ansible_hashivault_secret_engine_payload_n8x91xqz/ansible_hashivault_secret_engine_payload.zip/ansible/module_utils/hashivault.py\", line 154, in wrapper\n  File \"/tmp/ansible_hashivault_secret_engine_payload_n8x91xqz/ansible_hashivault_secret_engine_payload.zip/ansible/modules/hashivault/hashivault_secret_engine.py\", line 239, in hashivault_secret_engine\nTypeError: configure() got an unexpected keyword argument 'seal_wrap'\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

File: ansible/modules/hashivault/hashivault_secret_engine.py [hunk 1]
```diff
@@ -104,6 +115,7 @@ def main():
     argspec['options'] = dict(required=False, type='dict', default={})
     argspec['cas_required'] = dict(required=False, type='bool')
     argspec['max_versions'] = dict(required=False, type='int')
+    argspec['seal_wrap'] = dict(required=False, type='bool')
```
Comment (TerryHowe) [inline ansible/modules/hashivault/hashivault_secret_engine.py:118]: Maybe to be explicit, `default=False`

File: ansible/modules/hashivault/hashivault_secret_engine.py [hunk 2]
```diff
@@ -214,22 +227,55 @@ def hashivault_secret_engine(module):
...
         if backend == 'kv' or backend == 'kv-v2':
-            client.sys.enable_secrets_engine(backend, description=description, path=name, config=config,
-                                             options=options)
+            client.sys.enable_secrets_engine(
+                backend,
+                description=description,
+                path=name,
+                config=config,
+                options=options,
+                seal_wrap=seal_wrap
```
Comment (TerryHowe) [inline ansible/modules/hashivault/hashivault_secret_engine.py:236]: If seal_wrap is an enterprise only feature, you'll probably need to contruct kwargs and expand it here and only specify seal_wrap if true

**Review by TerryHowe — CHANGES_REQUESTED** [review]

File: ansible/modules/hashivault/hashivault_secret_engine.py (hunk 1 above)
Comment (emoryotott19) [reply ansible/modules/hashivault/hashivault_secret_engine.py]: Added this with commit 093c7eb

File: ansible/modules/hashivault/hashivault_secret_engine.py (hunk 2 above)
Comment (emoryotott19) [reply ansible/modules/hashivault/hashivault_secret_engine.py:236]: I think I see the issue now ... It isn't that it is an enterprise feature, it is that I was misusing the kvv2.configure method in hvac. That method is used for the kv specific configurations of the kv v2 secret engine. Adding [link to the hvac method](https://github.com/hvac/hvac/blob/9befaa4be80c88aa92c0ab69ba480d7c1d4009e5/hvac/api/secrets_engines/kv_v2.py#L18) for anyone who might run into this in the future. Thanks for your patience; still getting familiar with the code base :)

**Review by TerryHowe — APPROVED** [review]

**Pull request merged**